tools/setupResourceScripts/setup_tdr_resources.py

```diff
@@ -1,213 +1,278 @@
+      result = clients.jobs_api.retrieve_job_result(job_model.id)
+      raise f"Could not complete job with id {job_model.id}, got result {result}"
+    elif result.job_status=="succeeded":
+      print(f"Job {job_model.id} succeeded")
+      result = clients.jobs_api.retrieve_job_result(job_model.id)
+      return result
+    else:
+      raise "Unrecognized job state %s" % result.job_status
+
 
 # For dataset_ingest requests, each line in file is a json object
 # We need to convert to this to an array of json objects
 def convert_to_json_array(table_csv):
-    records = ""
-    for row in table_csv.readlines():
-        records += row.strip("\n") + ","
-    return json.loads("[" + records.strip(",") + "]")
+  records = ""
+  for row in table_csv.readlines():
+    records += row.strip("\n") + ","
+  return json.loads("[" + records.strip(",") + "]")
+
 
 def dataset_ingest_array(clients, dataset_id, dataset_to_upload):
-    for table in dataset_to_upload['tables']:
-        with open(os.path.join("files", dataset_to_upload["schema"],
-                               f"{table}.json")) as table_csv:
-            recordsArray = convert_to_json_array(table_csv)
-            if len(recordsArray) > 0:
-                ingest_request = {
-                    "format": "array",
-                    "records":recordsArray,
-                    "table": table
-                }
-                print(f"Ingesting data into {dataset_to_upload['name']}/{table}")
-                wait_for_job(clients, clients.datasets_api.ingest_dataset(dataset_id, ingest=ingest_request))
-            else:
-                print(f"Skipping ingest of {dataset_to_upload['name']}/{table} because it is empty")
-
-def create_billing_profile(clients, add_jade_stewards, cloud_platform):
+  for table in dataset_to_upload['tables']:
+    with open(os.path.join("files", dataset_to_upload["schema"],
+      f"{table}.json")) as table_csv:
+      recordsArray = convert_to_json_array(table_csv)
+      if len(recordsArray) > 0:
+        ingest_request = {
+          "format": "array",
+          "records": recordsArray,
+          "table": table
+        }
+        print(f"Ingesting data into {dataset_to_upload['name']}/{table}")
+        wait_for_job(clients,
+          clients.datasets_api.ingest_dataset(dataset_id, ingest=ingest_request))
+      else:
+        print(f"Skipping ingest of {dataset_to_upload['name']}/{table} because it is empty")
+
+
+def create_billing_profile(clients, add_jade_stewards, cloud_platform, billing_profile_file_name,
+                           azure_managed_app_name):
+  if not billing_profile_file_name:
     billing_profile_file_name = "billing_profile.json"
-    if cloud_platform == "azure":
-        billing_profile_file_name = "azure_billing_profile.json"
-    with open(os.path.join("files", billing_profile_file_name)) as billing_profile_json:
-        billing_profile_request = json.load(billing_profile_json)
-        profile_id = str(uuid.uuid4())
```

Comment:
Oh, wow, I didn't realize we were overwriting the profile set in the file with a random profile id.